TAO/TAO_IDL/ast/ast_expression.cpp

```diff
@@ -1725,6 +1811,32 @@ eval_kind (AST_Expression::AST_ExprValue *ev, AST_Expression::EvalKind ek)
 // Apply binary operators to an AST_Expression after evaluating
 // its sub-expressions.
 // Operations supported: '+', '-', '*', '/'
+template <typename Type>
+bool
+do_eval_bin_op (AST_Expression::ExprComb op, Type a, Type b, Type &result)
+{
+  switch (op)
+    {
+    case AST_Expression::EC_add:
+      result = a + b;
+      break;
+    case AST_Expression::EC_minus:
+      result = a - b;
+      break;
+    case AST_Expression::EC_mul:
+      result = a * b;
+      break;
+    case AST_Expression::EC_div:
+      if (!b) return true;
```

Comment:
Does this apply to both integer and floating point?  Should it return false if it fails?

Reply:
> Does this apply to both integer and floating point?

Yes, I think so.

> Should it return false if it fails?

`true` is failure in this case.

Reply:
> 
> 
> > Does this apply to both integer and floating point?
> 
> Yes, I think so.

Then should floating point allow dividing by zero resulting in NaN or +/-Inf?

> 
> > Should it return false if it fails?
> 
> `true` is failure in this case.

yikes... how? why?

Reply:
> Then should floating point allow dividing by zero resulting in NaN or +/-Inf?

[It didn't do that before](https://github.com/DOCGroup/ACE_TAO/blob/943dd53bd48bf4a16c02f98d217c59bbb4b78ea7/TAO/TAO_IDL/ast/ast_expression.cpp#L1882-L1887), but yes I could skip the check if it's a floating point.

> yikes... how? why?

I wrote like that because there has to be an action taken if there's a failure, but not for a success.